src/components/TheTools.vue

```diff
@@ -0,0 +1,619 @@
+                      sessions[index].records[marker.endN].distance!) /
+                    1000
+                  ).toFixed(2)
+                }}
+                km
+              </div>
+              <input
+                class="form-range"
+                type="range"
+                :min="0"
+                :max="sessions[index].records.length - 1"
+                v-model="marker.endN"
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="pt-3">
+      <div
+        class="row m-0"
+        style="cursor: pointer"
+        data-bs-toggle="collapse"
+        data-bs-target="#fieldsRemovalTarget"
+        aria-expanded="false"
+        aria-controls="fieldsRemovalTarget"
+      >
+        <div class="text-start p-0">
+          <label class="pe-1">Remove Fields</label>
+          <i class="fa-regular fa-circle-question" title="Show or Hide Help Text"></i>
+        </div>
+      </div>
+      <div class="collapse show" id="fieldsRemovalTarget">
+        <p>Select any field you wish to remove from the entire trackpoints.</p>
+      </div>
+      <div v-for="(item, index) in fieldsRemoverList" :key="index">
+        <div class="form-check" v-show="showFieldRemover(item.value)">
+          <input
+            class="form-check-input"
+            type="checkbox"
+            :id="item.value"
+            :value="item.value"
+            :disabled="encodeMode == EncodeMode.Unknown"
+            v-model="removeFields"
+          />
+          <label class="form-check-label" style="color: var(--color-text)" :for="item.value">
+            {{ item.label }}
+          </label>
+        </div>
+      </div>
+    </div>
+    <div class="pt-4">
+      <div class="row">
+        <div>
+          <button class="w-100 btn btn-success" @click="proceed" :disabled="!isValidToProceed">
+            Proceed
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { ActivityFile, Record, Session } from '@/spec/activity'
+import {
+  EncodeMode,
+  EncodeSpecifications,
+  FileType,
+  Manufacturer,
+  Marker,
+  Sport
+} from '@/spec/activity-service'
+import { shallowRef } from 'vue'
+
+export default {
+  props: {
+    manufacturers: { type: Array<Manufacturer>, required: true },
+    sports: { type: Array<Sport>, required: true },
+    activities: { type: Array<ActivityFile>, required: true },
+    sessions: { type: Array<Session>, required: true },
+    combinedRecord: Array<Record>
+  },
+  data() {
+    return {
+      manufacturerMap: shallowRef(new Map<number, Manufacturer>()),
+      sportMap: shallowRef(new Map<string, Sport>()),
+      EncodeMode: EncodeMode,
+      encodeMode: EncodeMode.Unknown,
+      selectedManufacturerProduct: null as unknown as any[],
+      manufacturerName: '',
+      manufacturerId: null as unknown as number | null,
```

Comment:
is it really unknown?

Reply:
it's just to make manufacturerId type is `number | null`, we we can't do it straightforwardly in data(), need to convert it to unknown first and cast it back